LoopKitUI/Views/Settings Editors/BasalRateScheduleEditor.swift

```diff
@@ -53,18 +53,16 @@ public struct BasalRateScheduleEditor: View {
     
     public init(
         viewModel: TherapySettingsViewModel,
-        mode: PresentationMode = .acceptanceFlow
+        onSave save: @escaping (BasalRateSchedule) -> Void
     ) {
         self.init(
             schedule: viewModel.therapySettings.basalRateSchedule,
             supportedBasalRates: viewModel.pumpSupportedIncrements!.basalRates ,
             maximumBasalRate: viewModel.therapySettings.maximumBasalRatePerHour,
             maximumScheduleEntryCount: viewModel.pumpSupportedIncrements!.maximumBasalScheduleEntryCount,
             syncSchedule: viewModel.syncPumpSchedule,
-            onSave: { newRates in
```

Comment:
Okay, but now you've removed the responsibility this had on actually doing the saving.  :).  If you want to do that, you'll need to wire up `saveBasalRates` (and all the other "saves").  Is that what you want to do?  Seems to me to be "convenient" you want both, no?

```
onSave: { newRates in
       viewModel.saveBasalRates(basalRates: newRates)
       save(newRates)
}
```

Otherwise this will break the settings screens.

Reply:
Hmm, I'm getting a comment that this is outdated. I made it so the closure could be passed in as an argument because the acceptance flow shouldn't save the setting into memory until it's finished.

Reply:
or...
```
  onSave: { newRates in 
       viewModel.saveBasalRates(basalRates: newRates)
      didSave()
}
```

then `init` just looks like:

```
    public init(
        viewModel: TherapySettingsViewModel,
        didSave: (() -> Void)? = nil
    ) {
```